cunumeric/runtime.py

```diff
@@ -241,45 +256,59 @@ def destroy(self) -> None:
             self._report_coverage()
         self.destroyed = True
 
-    def create_scalar(self, array: memoryview, dtype, shape=None, wrap=False):
+    def create_scalar(
+        self,
+        array: memoryview,
+        dtype: np.dtype[Any],
+        shape: Optional[NdShape] = None,
+    ) -> Future:
         data = array.tobytes()
         buf = struct.pack(f"{len(data)}s", data)
-        future = self.legate_runtime.create_future(buf, len(buf))
-        if wrap:
-            assert all(extent == 1 for extent in shape)
-            assert shape is not None
-            store = self.legate_context.create_store(
-                dtype,
-                shape=shape,
-                storage=future,
-                optimize_scalar=True,
-            )
-            result = DeferredArray(self, store, dtype=dtype)
-        else:
-            result = future
-        return result
+        return self.legate_runtime.create_future(buf, len(buf))
+
+    def create_wrapped_scalar(
+        self, array: memoryview, dtype: np.dtype[Any], shape: NdShape
+    ) -> DeferredArray:
+        future = self.create_scalar(array, dtype, shape)
+        assert all(extent == 1 for extent in shape)
+        store = self.legate_context.create_store(
+            dtype,
+            shape=shape,
+            storage=future,
+            optimize_scalar=True,
+        )
+        return DeferredArray(self, store, dtype=dtype)
```

Comment:
Split `create_scalar` and `create_wrapped_scalar` here in order to avoid unpleasant union return type.